build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/PDEState.java

```diff
@@ -656,10 +656,10 @@ private static Collection<String> querySystemPackages(IVMInstall vm, IExecutionE
 					try (ZipFile zip = new ZipFile(path.toFile())) {
 						// Collect names of all directories that contain a .class file
 						zip.stream().filter(e -> !e.isDirectory()).map(ZipEntry::getName) //
-								.filter(n -> n.endsWith(".class")) //$NON-NLS-1$
+								.filter(n -> n.endsWith(".class") && n.indexOf('/') > 0) //$NON-NLS-1$
```

Comment:
I assume an entry-name like `/MyClass.class` is not possible or at least not relevant.

Reply:
Is`/MyClass.class` not just a default package case? It is still a valid case I think.

Reply:
We don't need to add "default" package to system provided packages. 
We are talking about something that PDE needs to address via "import package". How PDE should import package `""` and how relevant that can be for anyone?

I wonder which "old" JRE Denis has and if that is something completely wrong JRE but automatically found & configured by the new "autodetect JRE installations" job.

Reply:
@dhuebner : would be nice if you could check that and which exact JRE (or something that looks like JRE) is causing the trouble.

Reply:
@iloveeclipse 
I think it is the bundled JRE version:
`-vm
/Users/dhuebner/Downloads/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.justj.openjdk.hotspot.jre.full.macosx.aarch64_17.0.8.v20230801-1951/jre/lib/libjli.dylib
`

Reply:
Moment, this is Java 17, we are supposed to see only Java < 8? 
@HannesWell : that is at least not what the original PR expected?

Reply:
> Is`/MyClass.class` not just a default package case? It is still a valid case I think.

But even if a class is in the jar root, I assume the entry is just named `MyClass.class`, i.e. without leading slash. But I don't know if that applies for all implementations (haven't checked the docs yet).

> We don't need to add "default" package to system provided packages. We are talking about something that PDE needs to address via "import package". How PDE should import package `""` and how relevant that can be for anyone?
> 
Yes exactly, that's not relevant for that reason.

> Moment, this is Java 17, we are supposed to see only Java < 8? @HannesWell : that is at least not what the original PR expected?

Yes, this code path is only entered if `JavaRuntime.isModularJava(vm)` returns false.
This only happens if the vm is not an `AbstractVMInstall`, `AbstractVMInstall.getJavaVersion()` returns null or a value below 1.8 respectively 8. Maybe the VMInstall type is special or not correctly configured?

Any way I think this change is fine to merge as it should not harm valid installations.
Nevertheless finding the root cause would be interesting for https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/294.

Reply:
> Nevertheless finding the root cause would be interesting for [eclipse-jdt/eclipse.jdt.debug#294](https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/294).

See https://github.com/eclipse-pde/eclipse.pde/issues/724#issuecomment-1693201112